### ISSUE TITLE
accountsexpires not retrieved

### ISSUE THREAD

**Issue by cluwong** [Adldap2/Adldap2#433]
- Adldap2 Version: 7.0.15
- LDAP Type: Active Directory
- PHP Version: 7.0

### Description:
$user->getAccountExpiry() does not return the correct value probably because 'accountexpires' is not included with query result.

### Steps To Reproduce:
Tested with and without select():

```php
public function getUser($id)
{
    return $provider->search()->users()->findBy('samaccountname', $id);
}

public function getUser($id)
{
    return $provider->search()->users()->select('accountexpires')->findBy('samaccountname', $id);
}
```

The only attributes returned, when select is specified are objectclass, objectcategory and dn.

Have tested from another site which uses Adldap2 V6 on an expired user account, 'accountexpires' exists via the following function call:

```php
public function getUser($id)
{
    return $p->search()->users()->where('samaccountname', '=', $id)->first();
}
```

**Comment by stevebauman** [comment]
Hi @cluwong,

That's strange, as I'm able to retrieve account expiry's without issue.

Just have some questions:

- Are you sure your running `v7.0.15`? (Just making sure)
- Are all attributes returned on the user account when you don't specify fields to select?

**Comment by cluwong** [comment]
Now that you ask, I'm not too sure about the minor version number... Adldap2 is installed via composer.  My app uses another package (https://github.com/edvler/yii2-adldap-module) which is a wrapper for Adldap2 and V7 is specified in that package's composer.

I'm not sure if all attributes are returned since I don't use all attributes. (My function does not select fields, i'm only adding select() for testing). Only noticed 'accountexpires' is missing because it is of of the attributes needed.

Comparing with V6, V7 returns less number of attributes when select is not specified. Eg on the same account 81 (V6) and 66 (V7).

**Comment by stevebauman** [comment]
Strange... Just to be safe, can you run a `composer update` on your application?

Are you using the same credentials with `v6` as you are `v7`?

**Comment by Albvadi** [comment]
Don´t close this issue by inactivity...
I think I have the same issue, but I need a few days to test it...

**Comment by stevebauman** [comment]
Ok thanks @Albvadi!

**Comment by cluwong** [comment]
Sorry for taking so long to get back to this as I was working on another project.

After taking a break from this project and getting back to it, I finally spotted something different in the config file. The new project was connecting to a different port, and after further investigation, 'accountexpires' is not cached in this directory - and some other attributes are not cached, hence I was getting less number of attributes in this project.
Sorry, it's not your code. The config/credential was copied & pasted and forgot that the port was changed sometimes during dev for testing and was too blind to notice the difference when comparing the config.
